packages/api/src/management.ts

```diff
@@ -0,0 +1,152 @@
+ * This is used when requesting an access token for the Management API.
+ */
+export const allScope = 'all';
+
+type ManagementApiReturnType = {
+  /**
+   * The API client for the Management API.
+   *
+   * This client is configured to use the provided client credentials
+   * and will automatically include the access token in requests.
+   */
+  apiClient: Client<paths>;
+  /**
+   * The client credentials instance used for authentication.
+   */
+  clientCredentials: ClientCredentials;
+};
+
+/**
+ * Creates a Management API client with the specified tenant ID and options.
+ *
+ * Before using this function, ensure that you have created a machine-to-machine application in
+ * Logto and granted it access to the Management API. See the documentation for more details:
+ *
+ * https://a.logto.io/m2m-mapi
+ *
+ * This function sets up the API client with the necessary authentication using client credentials.
+ * It will automatically handle token retrieval and renewal as needed.
+ *
+ * @param tenantId The tenant ID for which to create the Management API client. For OSS deployments,
+ * you can pass any string as the tenant ID, for example, 'default'.
+ * @param options The options for creating the Management API client, including client ID and secret.
+ * @returns An object containing the API client and client credentials instance.
+ * @example
+ * ```ts
+ * import { createManagementApi } from '@logto/api/management';
+ *
+ * // Logto Cloud example
+ * const { apiClient, clientCredentials } = createManagementApi('my-tenant-id', {
+ *   clientId: 'my-client-id',
+ *   clientSecret: 'my-client-secret',
+ * });
+ *
+ * // Use apiClient to make requests to the Management API
+ * const response = await apiClient.GET('/api/users');
+ * console.log(response.data);
+ * ```
+ *
+ * @example
+ * ```ts
+ * // OSS example
+ * const { apiClient, clientCredentials } = createManagementApi('default', {
+ *   clientId: 'my-client-id',
+ *   clientSecret: 'my-client-secret',
+ *   baseUrl: 'https://my-oss-logto-instance.com',
+ *   apiIndicator: 'https://default.logto.app/api',
+ * });
+ * ```
+ */
+export function createManagementApi(
+  tenantId: string,
+  options: CreateManagementApiOptions
+): ManagementApiReturnType {
+  const { clientId, clientSecret } = options;
+  const baseUrl = options.baseUrl ?? getBaseUrl(tenantId);
+  const apiIndicator = options.apiIndicator ?? getManagementApiIndicator(tenantId);
+  const clientCredentials = new ClientCredentials({
+    clientId,
+    clientSecret,
+    tokenEndpoint: `${baseUrl}/oidc/token`,
+    tokenParams: {
+      resource: apiIndicator,
+      scope: allScope,
+    },
+  });
+  const apiClient = createClient<paths>({
+    baseUrl,
+  });
+
+  apiClient.use({
+    async onRequest({ schemaPath, request }) {
+      if (schemaPath.includes('/.well-known/')) {
+        // Skip auth for well-known endpoints
+        return;
+      }
+      const { value, scope } = await clientCredentials.getAccessToken();
```

Comment:
Maybe we can add more detailed error messages on this error, particularly for new users who are unfamiliar with this process.

Reply:
continue discussion offline. will submit a new pull if needed.